client/src/services/user.js

```diff
@@ -0,0 +1,15 @@
+let isFetching = false;
+let user;
```

Comment:
Same thing with the custom hook here. The side effect in the function doesn't feel like the right way to do this; I'd make a custom hook with useState.